lib/ansible/modules/notification/irc.py

```diff
@@ -155,6 +155,16 @@
 
 from time import sleep
 
+def try_recv(sock, num=1024):
+    '''reads from socket and responds to ping if necessary'''
+    while True:
```

Comment:
This loop checks if any PING was sent (IRC heartbeat) and responds to that immediately. This may also be achieved by checking using something like
```
if ping:
    sock.send(...)
    ret=sock.recv(...)
```
but that would break if the server sent two pings.